ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/LedgerDB/OnDisk.hs

```diff
@@ -975,17 +995,23 @@ runDB standalone@DB{..} cmd =
             S.encode
             db
     go hasFS Restore = do
+        LedgerBackingStore old_db <- atomically . readTVar $ dbBackingStore
+        HD.bsClose old_db
         (initLog, db, _replayed, backingStore) <-
           initLedgerDB
             nullTracer
-            nullTracer
+            (Trace.Tracer $ \case
+               LMDBEvent e -> Trace.runTracer (dbTracer dbEnv) e
+               _ -> pure ()) -- TODO gross
```

Comment:
We have (at least) two options: trace `LMDB.TraceDb` events only, or `TraceEvent blk` as well. Seeing as this implementation seems to be intended to only trace `LMDB.TraceDb`, I'd propose to do only a minor refactoring to use an abastraction like `Trace.traceMaybe` or `Trace.squelchUnless`

Reply:
I see now that my last comment was a bit naive: `traceMaybe` is not sufficient because you still have to embed `Tracer m TraceDb` in `Tracer m TraceEvent`. I did find that the [`traceMaybe` function from the original `contra-variant` package](https://github.com/avieth/contra-tracer/blob/c8fc0aec6a57e3011aab907bdf48bbdb9a3224b2/src/Control/Tracer.hs#L217) would do exactly this embedding for us. Would it be worth it to upstream this function to the IOHK-fork of `contra-variant`?